# Oxford Nanopore tools

[Back to Bioinformatics Main
Menu](/kb3/Software/Bioinformatics/Bioinformatics/)

List of long-read software
[long-read-tools.org](http://long-read-tools.org)

## QC

### Porechop

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available: no

Porechop [homepage](https://github.com/rrwick/Porechop)

Porechop is a tool for finding and removing adapters from Oxford
Nanopore reads.

Adapters on the ends of reads are trimmed off, and when a read has an
adapter in its middle, it is treated as chimeric and chopped into
separate reads. Porechop performs thorough alignments to effectively
find adapters, even at low sequence identity.

`module load Porechop/0.2.4-goolfc-2017b-Python-3.6.3`

## Genome assembly

### Unicycler

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available: no

[Unicycler](https://github.com/rrwick/Unicycler) is an assembly pipeline
for bacterial genomes.

It circularises replicons without the need for a separate tool like
Circlator.

It can assemble Illumina-only read sets where it functions as a
SPAdes-optimiser. It can also assembly long-read-only sets (PacBio or
Nanopore) where it runs a miniasm+Racon pipeline. For the best possible
assemblies, give it both Illumina reads and long reads, and it will
conduct a hybrid assembly.

    module load Unicycler/0.4.6-intel-2017A-Python-3.5.2

### wtdbg2

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available:
[grace](https://github.tamu.edu/cmdickens/GCATemplates/blob/master/templates/grace/run_wtdbg2_2.5_fasta_assemble_grace.sh)
    [terra
(merge/assemble/polish)](https://github.tamu.edu/cmdickens/GCATemplates/blob/master/templates/terra/run_wtdbg2_2.3_merge_subreads_assemble_polish_terra.sh)

wtdbg2 [homepage](https://github.com/ruanjue/wtdbg2)

WTDBG: De novo assembler for long noisy sequences

Wtdbg2 is a de novo sequence assembler for long noisy reads produced by
PacBio or Oxford Nanopore Technologies (ONT). It assembles raw reads
without error correction and then builds the consensus from intermediate
assembly output.

`module spider wtdbg2`

## Assembly polishing

### Redundans

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available:
[ada](https://github.tamu.edu/cmdickens/GCATemplates/blob/master/templates/ada/run_redundans_0.13c_refbased_scaf_gapclose_pe_ada.sh)

Redundans [homepage](https://github.com/Gabaldonlab/redundans)

Redundans pipeline assists an assembly of heterozygous genomes. Program
takes as input assembled contigs, sequencing libraries and/or reference
sequence and returns scaffolded homozygous genome assembly. Final
assembly should be less fragmented and with total size smaller than the
input contigs. In addition, Redundans will automatically close the gaps
resulting from genome assembly or scaffolding.

`module load Redundans/0.13c-intel-2017b-Python-2.7.14`

### Racon

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available: no

Consensus module for raw de novo DNA assembly of long uncorrected reads.

Racon is intended as a standalone consensus module to correct raw
contigs generated by rapid assembly methods which do not include a
consensus step.

`module load Racon/1.3.1-GCCcore-6.3.0`

### Nanopolish

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available: no

Nanopolish [homepage](https://github.com/jts/nanopolish)

Software package for signal-level analysis of Oxford Nanopore sequencing
data. Nanopolish can calculate an improved consensus sequence for a
draft genome assembly, detect base modifications, call SNPs and indels
with respect to a reference genome and more.

`module load nanopolish/0.11.0-intel-2017A-Python-2.7.12`

## Sequence alignments

### Minimap2

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available:
[grace](https://github.tamu.edu/cmdickens/GCATemplates/blob/master/templates/grace/run_minimap2_2.23_pacbio_sort_grace.sh)

Minimap2 [homepage](https://github.com/lh3/minimap2)

Minimap2 is a versatile sequence alignment program that aligns DNA or
mRNA sequences against a large reference database.

`module spider minimap2`

### LAST

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available: no

LAST [homepage](http://last.cbrc.jp/)

` module spider LAST`

LAST finds similar regions between sequences. LAST copes more
efficiently with repeat-rich sequences (e.g. genomes).

For example: it can align reads to genomes without repeat-masking,
without becoming overwhelmed by repetitive hits.

### Nanopolish

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available: no

Nanopolish [homepage](https://github.com/jts/nanopolish)

Software package for signal-level analysis of Oxford Nanopore sequencing
data. Nanopolish can calculate an improved consensus sequence for a
draft genome assembly, detect base modifications, call SNPs and indels
with respect to a reference genome and more.

`module load nanopolish/0.11.0-intel-2017A-Python-2.7.12`

## Simulate Reads

### NanoSim

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available:
[ada](https://github.tamu.edu/cmdickens/GCATemplates/blob/master/templates/ada/run_nanosim_2.2.0_simulator_ada.sh)

NanoSim [homepage](https://github.com/bcgsc/NanoSim)

NanoSim is a fast and scalable read simulator that captures the
technology-specific features of ONT data, and allows for adjustments
upon improvement of nanopore sequencing technology.

`module load NanoSim/2.2.0-intel-2017A-Python-2.7.12`
