# ChIP-seq

[Back to Bioinformatics Main
Menu](/kb3/Software/Bioinformatics/Bioinformatics/)

## MACS

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available:
[ada](https://github.tamu.edu/cmdickens/GCATemplates/blob/master/templates/ada/run_macs_1.4.2_test_control_ada.sh)

MACS [homepage](http://liulab.dfci.harvard.edu/MACS/)

` module spider MACS`

Model-based Analysis of ChIP-Seq data.

## MACS2

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available: no

MACS2 [homepage](https://pypi.python.org/pypi/MACS2)

` module spider MACS2`

Model-based Analysis of ChIP-Seq data.

## SICER

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available:
[ada](https://github.tamu.edu/cmdickens/GCATemplates/blob/master/templates/ada/run_sicer_1.1_test_control_ada.sh)

SICER [homepage](http://home.gwu.edu/~wpeng/Software.htm)

` module spider SICER`

A clustering approach for identification of enriched domains from
histone modification ChIP-Seq data

## ChIP-Seq

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available: no

ChIP-Seq [homepage](http://chip-seq.sourceforge.net/)

` module spider ChIP-Seq`

The ChIP-Seq software provides methods for the analysis of ChIP-seq data
and other types of mass genome annotation data. The most common analysis
tasks include positional correlation analysis, peak detection, and
genome partitioning into signal-rich and signal-depleted regions.

## PeakRanger

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available: no

PeakRanger [homepage](http://ranger.sourceforge.net/)

` module spider PeakRanger`

A multi-purpose ultrafast peak caller for ChIP Seq data

## PeakSplitter

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available: no

PeakSplitter [homepage](http://www.ebi.ac.uk/research/bertone/software)

` module spider PeakSplitter`

Subdivision of ChIP-seq/ChIP-chip regions into discrete signal peaks

## BroadPeak

[GCATemplates](/kb3/Software/useful-tools/SW@GCATemplates/ "wikilink") available: no

BroadPeak
[homepage](http://jordan.biology.gatech.edu/page/software/broadpeak/)

` module spider BroadPeak`

BroadPeak broad peak calling algorithm for diffuse ChIP-seq datasets.
